x-pack/plugins/fleet/server/services/agent_policies/bump_agent_policies_task.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { Logger } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { v4 as uuidv4 } from 'uuid';
+import { uniq } from 'lodash';
+
+import { DEFAULT_SPACE_ID } from '@kbn/spaces-plugin/common';
+
+import { agentPolicyService, appContextService } from '..';
+import { runWithCache } from '../epm/packages/cache';
+import { getPackagePolicySavedObjectType } from '../package_policy';
+import { mapPackagePolicySavedObjectToPackagePolicy } from '../package_policies';
+import type { PackagePolicy, PackagePolicySOAttributes } from '../../types';
+import { normalizeKuery } from '../saved_object';
+import { SO_SEARCH_LIMIT } from '../../constants';
+
+const TASK_TYPE = 'fleet:bump_agent_policies';
+
+export function registerBumpAgentPoliciesTask(taskManagerSetup: TaskManagerSetupContract) {
+  taskManagerSetup.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Fleet Bump policies',
+      timeout: '5m',
+      maxAttempts: 3,
+      createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+        let cancelled = false;
+        return {
+          async run() {
+            if (cancelled) {
+              throw new Error('Task has been cancelled');
+            }
+
+            await runWithCache(async () => {
+              await _updatePackagePoliciesThatNeedBump(appContextService.getLogger(), cancelled);
```

Comment:
This isn't going to work, to handle being cancelled **while** it's running.   `cancelled` will always be false here.

Here's an example of handling this correctly; continue to capture it locally, but provide a function that test's it, and pass the function to the "inner" functions rather than the value:

https://github.com/elastic/kibana/blob/ebd3f0d69e21b84f7f65e5f57ef0afcffc2f808a/x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/tasks/risk_scoring_task.ts#L392-L408